hrms/hr/doctype/goal/goal.py

```diff
@@ -81,6 +81,8 @@ def validate_progress(self):
 			frappe.throw(_("Goal progress percentage cannot be more than 100."))
 
 	def set_status(self, status=None):
+		if self.status == "Closed":
+			return
 		if self.status != "Archived":
```

Comment:
```suggestion
		if self.status not in ["Archived", "Closed"]:
```
Or you could also write a combined early return
if self.status in ["Archived", "Closed"]: return

and then unindent the block